### PR TITLE
Introduce ConnectorMetadata and Connector

### LINKED ISSUE #5737: Metadata API

### PR DESCRIPTION
**Pull request by DorianZheng**
Signed-off-by: dorianzheng <xingzhengde72@gmail.com>

## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #5737 #5738 

## Problem Summary(Required) ：

Related with https://github.com/StarRocks/starrocks/issues/5062

Introduce ConnectorMetadata and Connector with some simple API

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.connector;
+
+import com.google.common.collect.Lists;
+import com.starrocks.catalog.Table;
+
+import java.util.List;
+
+public interface ConnectorMetadata {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java:10]: it should call Metadata, because 'LocalMetastore' need implement it.

Comment (DorianZheng) [reply fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java:10]: Metadata is a general name, besides ConnectorMetadata, there are going to have few `-Metadata`, so we need a more specific name. Local table is going to be integrated to Connector in the future, so `LocalMetastore` is ok to derived from `ConnectorMetadata`

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 3 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ConnectorMetadata.java|0|3|00.00%|

**Review by imay — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java (hunk 1 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java:10]: agree, but It still sounds a little strange😓

**Review by caneGuy — APPROVED** [review]

**Pull request merged**